client/src/components/Header.tsx

```diff
@@ -1,5 +1,51 @@
+import React, { useState, useEffect } from "react";
+
 const Header = () => {
-  return <div>Header</div>;
+  const [client_id, setClientId] = useState(Date.now());
+  const [ws, setWs] = useState<WebSocket | null>(null);
+
+  useEffect(() => {
+    const newWs = new WebSocket(`ws://localhost:8000/ws/${client_id}`);
```

Comment:
Can we not hard code the address? Is there an easy way to get the URL base in JS/react?

Reply:
```suggestion
    const basename = `${window.location.hostname}:${window.location.port}` 
    const newWs = new WebSocket(`ws://${basename}ws/${client_id}`);
```